src/Client.php

```diff
@@ -337,6 +337,26 @@ public function waitFor(string $locator, int $timeoutInSecond = 30, int $interva
         return $this->crawler = $this->createCrawler();
     }
 
+    /**
+     * @param string $locator
+     * @param int $timeoutInSecond
+     * @param int $intervalInMillisecond
+     * @throws NoSuchElementException
+     * @throws TimeoutException
+     */
+    public function waitForVisibility(string $locator, int $timeoutInSecond = 30, int $intervalInMillisecond = 250)
+    {
+        $locator = trim($locator);
+
+        $by = '' === $locator || '/' !== $locator[0]
+            ? WebDriverBy::cssSelector($locator)
+            : WebDriverBy::xpath($locator);
```

Comment:
These lines are identical to those starting the `waitFor()` method. In this case of this method and `waitFor()` this logic always needs to be present and if it changes for one it should change for the other. 

Can this be factored out to a private method called by both `waitForVisibility()` and `waitFor()`?

Reply:
Off the top of my head:

```php

private function createWebDriverByFromLocator(string $locator): WebDriverBy
{
        $locator = trim($locator);

        return '' === $locator || '/' !== $locator[0]
            ? WebDriverBy::cssSelector($locator)
            : WebDriverBy::xpath($locator);
}

Reply:
Thanks @webignition i added this function and use it in both `waitForVisibility()` and `waitFor()` .